cursor.go

```diff
@@ -0,0 +1,15 @@
+package engo
+
+// Cursor is a reference to standard GLFW-cursors, to be used in conjunction with `SetCursor`
+type Cursor uint8
+
+const (
+	// CursorNone can be used to reset the cursor
+	CursorNone = iota
```

Comment:
Add explicit type to constant definitions. This will both make the documentation more pleasant as they are grouped with the relevant type, and improve type checking, as they would no longer be considered untyped constants.

```diff
-CursorNone = iota
+CursorNone Cursor = iota